src/java.base/share/classes/java/lang/template/StringTemplate.java

```diff
@@ -0,0 +1,313 @@
+     *         than values list size
+     * @throws NullPointerException if fragments is null or values is null or if any fragment is null.
+     *
+     * @implNote Contents of both lists are copied to construct immutable lists.
+     */
+    public static StringTemplate of(List<String> fragments, List<?> values) {
+        Objects.requireNonNull(fragments, "fragments must not be null");
+        Objects.requireNonNull(values, "values must not be null");
+        if (values.size() + 1 != fragments.size()) {
+            throw new IllegalArgumentException(
+                    "fragments list size is not one more than values list size");
+        }
+        return TemplateSupport.of(fragments, values);
+    }
+
+    /**
+     * Creates a string that interleaves the elements of values between the
+     * elements of fragments.
+     *
+     * @param fragments  list of String fragments
+     * @param values     list of expression values
+     *
+     * @return String interpolation of fragments and values
+     *
+     * @throws IllegalArgumentException if fragments list size is not one more
+     *         than values list size
+     * @throws NullPointerException fragments or values is null or if any of the fragments is null
+     */
+    public static String interpolate(List<String> fragments, List<?> values) {
+        Objects.requireNonNull(fragments, "fragments must not be null");
+        Objects.requireNonNull(values, "values must not be null");
+        int fragmentsSize = fragments.size();
+        int valuesSize = values.size();
+        if (fragmentsSize != valuesSize + 1) {
+            throw new IllegalArgumentException("fragments must have one more element than values");
+        }
+        return TemplateSupport.interpolate(fragments, values);
+    }
+
+    /**
+      * Combine one or more {@link StringTemplate StringTemplates} to produce a combined {@link StringTemplate}.
+      * {@snippet :
+      * StringTemplate st = StringTemplate.combine(RAW."\{a}", RAW."\{b}", RAW."\{c}");
+      * assert st.interpolate().equals(RAW."\{a}\{b}\{c}");
+      * }
+      *
+      * @param sts  one or more {@link StringTemplate}
+      *
+      * @return combined {@link StringTemplate}
+      *
+      * @throws NullPointerException if sts is null or if any of the elements are null
+      * @throws RuntimeException if sts has zero elements
+      */
+    public static StringTemplate combine(StringTemplate... sts) {
+        return TemplateSupport.combine(sts);
+    }
+
+    /**
+     * Interpolation template processor instance.
+     * Example: {@snippet :
+     * int x = 10;
+     * int y = 20;
+     * String result = STR."\{x} + \{y} = \{x + y}"; // @highlight substring="STR"
+     * }
+     * @implNote The result of interpolation is not interned.
+     */
+    public static final StringProcessor STR = StringTemplate::interpolate;
+
+    /**
+     * No-op template processor. Used to highlight that non-processing of the StringTemplate
```

Comment:
It's a reasonable suggestion. This processor would used rarely, so calling it IDENTITY (more than 3 characters) is not an impediment. However, IDENTITY doesn't really speak to what it does. RAW was chosen primarily because it is what other languages use. (I had used NOOP  earlier.)